Add option in `postgres_session` resource to establish socket connection (#5664)
chef/inspec: docs-chef-io/content/inspec/resources/postgres_session.md

A `postgres_session` resource block declares the username and password to use for the session, and then the command to be run:

    # Create a PostgreSQL session:
    sql = postgres_session('username', 'password', 'host', 'port', 'socketpath')

    # default values:
    #   username: 'postgres'
    #   host: 'localhost'
    #   port: 5432
    #   socketpath (optional): nil

    # Run an SQL query with an optional database to execute
    sql.query('sql_query', ['database_name'])`

A full example is:

    sql = postgres_session('username', 'password', 'host', 'port', 'socketpath')
    describe sql.query('SELECT * FROM pg_shadow WHERE passwd IS NULL;') do
      its('output') { should eq '' }
    end

where
  - `its('output') { should eq '' }` compares the results of the query against the expected result in the test

>   - `socketpath` is an optional parameter. It can be used to establish socket connection with postgres by specifying one of the postgres unix domain sockets path. Only supported for unix based platforms.

Reviewer: ```suggestion
  - `socketpath` is an optional parameter. Use `socketpath` to establish a socket connection with Postgres by specifying one of the Postgres Unix domain socket paths. Only supported on Unix-based platforms.
```
